### PR TITLE
[Feature] Remove remote grant during deauthorization if possible

### PR DESCRIPTION
**Pull request by mfraezz**
[#OSF-5365]

Purpose
=======
Addons that follow the [OAuth RFC](http://tools.ietf.org/html/rfc6749) most closely provide an endpoint to revoke the oauth grant on their end, invalidating any access tokens that may have been discovered by a third-party. We should make use of this feature for addons with an API supporting it.

Changes
=======
* Method and call added to base class that will `pass` when tokens should be revoked.
* Above method overridden in addons that support token revocation (and have been migrated to the `ExternalAccount` architecture already.)
* #4853 has had this method preemptively overriden on the `GitHubUserSettings` model.

Side Effects
==========
None

### CODE REVIEW

**Comment by samanehsan** [comment]
Code looks good! :tropical_fish: But, when I tested this out for dropbox I still saw the app listed in my dropbox account settings.

**Comment by samanehsan** [comment]
According to @mfraezz: 
>So the grant revocation ticket was slightly incorrect. As far as I can tell, none of our addons support complete revocation though their API, just token disabling. Which is actually what's being done.

**Comment by samanehsan** [comment]
Tested out disabling tokens for box and dropbox and it is working as expected. After disabling the dropbox and box addons, I cannot make any more requests with the disabled tokens. :+1:

**Comment by samanehsan** [comment]
Done with code review! :whale:

File: website/addons/box/model.py [hunk 1]
```diff
@@ -59,6 +60,21 @@ class BoxUserSettings(AddonOAuthUserSettingsBase):
     oauth_provider = Box
     serializer = BoxSerializer
 
+    def revoke_remote_oauth_access(self, external_account):
+        try:
+            # TODO: write client for box, stop using third-party lib
```
Comment (sloria) [inline website/addons/box/model.py]: Why rewrite the client?

File: tests/test_addons_oauth.py [hunk 2]
```diff
@@ -294,6 +294,23 @@ def test_grant_metadata(self):
             }
         )
 
+    @mock.patch('tests.test_addons_oauth.MockUserSettings.revoke_remote_oauth_access')
+    @mock.patch('framework.auth.core._get_current_user')
+    def test_revoke_remote_access_called(self, mock_decorator, mock_revoke):
```
Comment (sloria) [inline tests/test_addons_oauth.py]: Minor: `mock_decorator` isn't a descriptive name; `mock_get_current_user` would make more sense.

**Pull request merged**